src/date-time.ts

```diff
@@ -0,0 +1,553 @@
+            .ensure(t => t.matchesFormat("####-##-##"));
+
+        given(timeValue, "timeValue").ensureHasValue().ensureIsString()
+            .ensure(t => t.matchesFormat("##:##"));
+
+        given(zone, "zone").ensureHasValue().ensureIsString();
+
+        const dateTimeString = `${dateValue} ${timeValue}`;
+
+        return new DateTime({
+            value: dateTimeString,
+            zone
+        });
+    }
+
+    public static min(dateTime1: DateTime, dateTime2: DateTime): DateTime
+    {
+        given(dateTime1, "dateTime1").ensureHasValue().ensureIsType(DateTime);
+        given(dateTime2, "dateTime2").ensureHasValue().ensureIsType(DateTime);
+
+        if (dateTime1.valueOf() < dateTime2.valueOf())
+            return dateTime1;
+
+        return dateTime2;
+    }
+
+    public static max(dateTime1: DateTime, dateTime2: DateTime): DateTime
+    {
+        given(dateTime1, "dateTime1").ensureHasValue().ensureIsType(DateTime);
+        given(dateTime2, "dateTime2").ensureHasValue().ensureIsType(DateTime);
+
+        if (dateTime1.valueOf() > dateTime2.valueOf())
+            return dateTime1;
+
+        return dateTime2;
+    }
+
+    public static validateDateTimeFormat(value: string): boolean
+    {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        if (value == null || value.isEmptyOrWhiteSpace())
+            return false;
+
+        return LuxonDateTime.fromFormat(value, DateTime._format).isValid;
+    }
+
+    public static validateDateFormat(value: string): boolean
+    {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        if (value == null || value.isEmptyOrWhiteSpace())
+            return false;
+
+        return LuxonDateTime.fromFormat(value, "yyyy-MM-dd").isValid;
+    }
+
+    public static validateTimeFormat(value: string): boolean
+    {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        if (value == null || value.isEmptyOrWhiteSpace())
+            return false;
+
+        return LuxonDateTime.fromFormat(value, "HH:mm").isValid;
+    }
+
+    public static validateTimeZone(zone: string): boolean
+    {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        if (zone == null || zone.isEmptyOrWhiteSpace())
+            return false;
+
+        try
+        {
+            DateTime._validateZone(zone);
+        }
+        catch
+        {
+            return false;
+        }
+
+        return LuxonDateTime.now().setZone(zone).isValid;
+    }
+
+
+    private static _validateZone(zone: string): void
+    {
+        if (zone.toLowerCase() === "utc")
+            return;
+
+        given(zone, "zone")
+            .ensureWhen(
+                zone.toLowerCase() === "local",
+                _ => false,
+                "should not use local zone")
+            .ensureWhen(
+                IANAZone.isValidSpecifier(zone),
```

Comment:
Why do you have this? doesn't luxon validate this?

Reply:
I was being specific about the error

Reply:
Na don't need this

Reply:
Imagine the list of all the zone, and it is probably iterating over all of them to find if it's in there or not. Now that is happening 2 times once here and once in creation of DateTime